nixos/modules/security/apparmor/includes.nix

```diff
@@ -0,0 +1,301 @@
+    ${etcRule {path="alsa/conf.d"; trail="/";}}
+    ${etcRule {path="alsa/conf.d"; trail="/*";}}
+    ${etcRule "openal/alsoft.conf"}
+    ${etcRule "wildmidi/wildmidi.conf"}
+  '';
+  "abstractions/authentication" = ''
+    include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/authentication"
+    # Defined in security.pam
+    include <abstractions/pam>
+    ${etcRule "nologin"}
+    ${etcRule "securetty"}
+    ${etcRule {path="security"; trail="/*";}}
+    ${etcRule "shadow"}
+    ${etcRule "gshadow"}
+    ${etcRule "pwdb.conf"}
+    ${etcRule "default/passwd"}
+    ${etcRule "login.defs"}
+  '';
+  "abstractions/base" = ''
+     include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/base"
+     r ${pkgs.stdenv.cc.libc}/share/locale/**,
+     r ${pkgs.stdenv.cc.libc}/share/locale.alias,
+     ${lib.optionalString (pkgs.glibcLocales != null) "r ${pkgs.glibcLocales}/lib/locale/locale-archive,"}
+     ${etcRule "localtime"}
+     r ${pkgs.tzdata}/share/zoneinfo/**,
+     r ${pkgs.stdenv.cc.libc}/share/i18n/**,
+  '';
+  "abstractions/bash" = ''
+    include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/bash"
+    # system-wide bash configuration
+    ${etcRule "profile.dos"}
+    ${etcRule "profile"}
+    ${etcRule "profile.d"}
+    ${etcRule {path="profile.d"; trail="/*";}}
+    ${etcRule "bashrc"}
+    ${etcRule "bash.bashrc"}
+    ${etcRule "bash.bashrc.local"}
+    ${etcRule "bash_completion"}
+    ${etcRule "bash_completion.d"}
+    ${etcRule {path="bash_completion.d"; trail="/*";}}
+    # bash relies on system-wide readline configuration
+    ${etcRule "inputrc"}
+    # bash inspects filesystems at startup
+    # and /etc/mtab is linked to /proc/mounts
+    @{PROC}/mounts
+
+    # run out of /etc/bash.bashrc
+    ${etcRule "DIR_COLORS"}
+  '';
+  "abstractions/cups-client" = ''
+    include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/cpus-client"
+    ${etcRule "cups/cups-client.conf"}
+  '';
+  "abstractions/consoles" = ''
+     include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/consoles"
+  '';
+  "abstractions/dbus-session-strict" = ''
+    include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/dbus-session-strict"
+    ${etcRule "machine-id"}
+  '';
+  "abstractions/dconf" = ''
+    include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/dconf"
+    ${etcRule {path="dconf"; trail="/**";}}
+  '';
+  "abstractions/dri-common" = ''
+    include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/dri-common"
+    ${etcRule "drirc"}
+  '';
+  # The config.fonts.fontconfig NixOS module adds many files to /etc/fonts/
+  # by symlinking them but without exporting them outside of its NixOS module,
+  # those are therefore added there to this "abstractions/fonts".
+  "abstractions/fonts" = ''
+    include "${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/fonts"
+    ${etcRule {path="fonts"; trail="/**";}}
+  '';
+  "abstractions/gnome" = ''
```

Comment:
Perhaps this would look better as a nix list of attribute sets that are then translated once with once function instead of doing string interpolation over and over again?

Reply:
Indeed `include` and `etcRule` calls could be factorized by generating from a list of various attributes like `includeUpstream`, `textBefore`, `etcRules`, `textAfter`, etc., but what would be the benefit of rigidifyng this code? and what is the problem with repeated string interpolations?

Reply:
Or maybe I can just factorize the places where there are many `etcRule`s, I'll try and you'll tell me.